CONTRIBUTING.md

```diff
@@ -45,9 +45,87 @@ include your changes.
   the name of the thing you're changing in at the beginning of the message,
   followed by a colon: the plugin name for plugins, "docs" for documentation
   files, "coretasks" for `coretasks.py`, "db" for the database feature, etc.
-* Python files should always have  `from __future__ import generator_stop`
+* Python files should always have `from __future__ import annotations`
   as the first line after the module docstring.
 
+Running from source
+-------------------
+
+Sopel is your typical Python project: you can install it from source, edit
+the code using standard tooling, and take advantage of your favorite code
+editor.
+
+Assuming you are using standard tools:
+
+* Create a virtualenv.
+  * You can use `virtualenv` directly, `virtualenvwrapper`, or any tool that
```

Comment:
May want to also mention `venv` here for a stdlib option, and a link to the [PyPA](https://packaging.python.org/en/latest/guides/installing-using-pip-and-virtual-environments/#creating-a-virtual-environment) guide might be helpful for contributors who aren't already familiar.

Reply:
Hm, the PyPA guide says "virtualenv (for Python 2)" 🤔

Are `virtualenv` and `venv` close enough to feature parity that we can just recommend the stdlib option (`venv`) for newbies "or any other appropriate tool" if someone already has a preferred one?

Reply:
The main difference is that `venv` provides an environment for whatever Python it's running in, and `virtualenv` has more ambition for managing multiple versions.

I would definitely be inclined towards recommending "`venv` or other tools" here, the users who want something other than the standard option probably don't need the docs to guide them to it anyway.

Reply:
I'd rather say:

> You can use `virtualenv` directly (or the `venv` built-in), `virtualenvwrapper` to manage multiple projects, or any tool that [...]